### ISSUE TITLE
db.schema.hasTable('') returns true

### ISSUE THREAD

**Issue by capaj** [tgriesser/knex#2065]
This may be some weird behaviour of mysql I don't know about, but shouldn't this method return false?
```javascript
const hasTable = await db.schema.hasTable('')
  if (hasTable) {
    console.log('hasTable: ', hasTable)
```

debug output:
```
  knex:query show tables like ? +2ms
  knex:bindings [ '' ] +0ms
  knex:client releasing connection to pool: __knexUid1 +10ms
  knex:pool INFO pool mysql:mysql:client0 - dispense() clients=0 available=3 +1ms
hasTable:  true

```

On Knex 0.13.0

**Comment by elhigu** [comment]
looks like it is not exact match, that hasTable does on mysql:

```sql
MariaDB [knex_test]> show tables like '';
+----------------------------+
| Tables_in_knex_test ()     |
+----------------------------+
| 10_test_table              |
| BatchInsert                |
| accounts                   |
| batchInsertDuplicateKey    |
| bool_test                  |
| catch_test                 |
| charset_collate_test       |
| composite_key_test         |
| datatype_test              |
| invalid_inTable_param_test |
| knex_migrations            |
| knex_migrations_lock       |
| person                     |
| primarytest                |
| renameColTest              |
| taabl                      |
| test_default_table         |
| test_default_table2        |
| test_table                 |
| test_table_two             |
+----------------------------+
20 rows in set (0.00 sec)

MariaDB [knex_test]> 
```

I would consider this as a knex bug. Knex should check that exact match was found in the returned table list or change query to be something that throws an error if table doesn't exist like shown here  http://stackoverflow.com/a/8829109/360060

**Comment by elhigu** [comment]
Fixed via #2097
